core/cog-view.c

```diff
@@ -454,3 +454,51 @@ cog_view_get_viewport(CogView *self)
     g_return_val_if_fail(COG_IS_VIEW(self), NULL);
     return g_weak_ref_get(&((CogViewPrivate *) cog_view_get_instance_private(self))->viewport);
 }
+
+/**
+ * cog_view_is_visible:
+ * @self: A view.
+ *
+ * Gets if the view is visible inside of the viewport where the view
+ * is attached.
+ *
+ * Returns: True if the view is visible in its viewport and false in any
+ * other case.
+ *
+ * Since: 0.20
+ */
+gboolean
+cog_view_is_visible(CogView *self)
+{
+    g_return_val_if_fail(COG_IS_VIEW(self), false);
+
+    g_autoptr(CogViewport) viewport = cog_view_get_viewport(self);
+    g_return_val_if_fail(viewport != NULL, false);
+
+    return self == cog_viewport_get_visible_view(viewport);
+}
+
+/**
+ * cog_view_set_visible:
+ * @self: A view.
+ *
+ * Set the view as the one visible inside of the viewport where the view is
+ * attached.
+ *
+ * Returns: True if the view succesfully set as visible in its viewport and
+ * false in any other case.
```

Comment:
Maybe the following is easier to parse (well, it is for me):

```suggestion
 * Set the view as visible in its viewport.
 *
 * Returns: %TRUE if the view was set as visible, or %FALSE if the view
 *    does is not attached to a viewport.
```